applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1467,51 +1525,59 @@ int unicast_client_discover(struct bt_conn *conn, enum unicast_discover_dir dir)
 	return ret;
 }
 
-int unicast_client_start(void)
+int unicast_client_start(uint8_t cig_index)
 {
 	int ret;
 	struct bt_cap_unicast_audio_start_stream_param
 		cap_stream_params[CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SRC_COUNT +
 				  CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SNK_COUNT];
 	static struct bt_cap_unicast_audio_start_param param;
 
+	if (cig_index >= CONFIG_BT_ISO_MAX_CIG) {
+		LOG_ERR("Trying to start CIG %d out of %d", cig_index, CONFIG_BT_ISO_MAX_CIG);
+		return -EINVAL;
+	}
+
 	param.stream_params = cap_stream_params;
 	param.count = 0;
 	param.type = BT_CAP_SET_TYPE_AD_HOC;
 
-	for (int i = 0; i < ARRAY_SIZE(unicast_servers); i++) {
+	for (int i = 0; i < ARRAY_SIZE(unicast_servers[cig_index][0]); i++) {
 		uint8_t state;
 
```

Comment:
Consider a variable for `unicast_servers[cig_index][0]` that's accessed eight times